src/parse.rs

```diff
@@ -297,29 +292,124 @@ pub fn parse_ids(
     }
 }
 
+pub fn parse_idle<'a>(
+    mut lines: &'a [u8],
+    unsolicited: &mut UnsolicitedResponseSender,
+) -> Result<bool> {
+    while !lines.is_empty() {
+        let resp = imap_proto::parse_response(lines);
+        match resp {
+            Ok((
+                _rest,
+                Response::Done {
+                    status,
+                    information,
+                    ..
+                },
+            )) => {
+                match status {
+                    imap_proto::Status::Ok => return Ok(true),
+                    imap_proto::Status::Bad => {
+                        return Err(Error::Bad(information.unwrap_or("").to_owned()))
+                    }
+                    imap_proto::Status::No => {
+                        return Err(Error::Bad(information.unwrap_or("").to_owned()))
+                    }
+                    _ => unreachable!(),
+                }
+            }
+            Ok((rest, data)) => {
+                lines = rest;
+                if let Some(resp) = handle_unilateral(data, unsolicited) {
```

Comment:
We should probably rename `handle_unilateral` to `try_handle_unilateral` to make it clearer that `Some` is returned if it wasn't handled. That's orthogonal to your change though :)